code/controllers/TaxonomyDirectoryController.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * Class TaxonomyDirectoryController
+ *
+ * Controller for returning a list of pages tagged with a specific Taxonomy Term
+ */
+class TaxonomyDirectoryController extends Page_Controller
+{
+
+    private static $allowed_actions = array(
+        'index'
+    );
+
+    public function index(SS_HTTPRequest $HTTPRequest)
+    {
+        //look up the term to avoid sql injection by using the ORM
+        $term = TaxonomyTerm::get()->filter(['URLSegment' => $HTTPRequest->param('ID')])->first();
+
+        if ($term) {
+
+            $title = $term->Name;
+
+            $results = Page::get()
+                ->innerJoin(
+                    'BasePage_Terms',
+                    '"Page"."ID"="BasePage_Terms"."BasePageID"')
+                ->innerJoin(
+                    'TaxonomyTerm',
+                    "\"BasePage_Terms\".\"TaxonomyTermID\"=\"TaxonomyTerm\".\"ID\" AND \"TaxonomyTerm\".\"ID\" = '$term->ID'");
+        } else {
+           //no term found return null results
```

Comment:
Question: should this throw a 404?

Reply:
This is more like a search results page so would rather return a "Sorry no results" type of message.

Reply:
ok